packages/flutter/test/semantics/semantics_update_test.dart

```diff
@@ -0,0 +1,227 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+import 'dart:ui' as ui;
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  SemanticsUpdateTestBinding();
+
+  testWidgets('Semantics update does not send update for merged nodes.', (WidgetTester tester) async {
+    final SemanticsHandle handle = tester.ensureSemantics();
+    await tester.pumpWidget(
+      MaterialApp(
+        home: MergeSemantics(
+          child: Semantics(
+            label: 'outer',
+            textField: true,
+            // This merge semantics node should not be part of the semantics
+            // update because it is under another merge semantics.
+            child: MergeSemantics(
```

Comment:
Can you achieve the same effect by removing this MergeSemantics and setting `container: true` on the semantics node below? Seems slightly simpler then the nested merging...